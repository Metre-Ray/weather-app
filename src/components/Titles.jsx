import React from 'react';
import styles from './Titles.module.css';

const Titles = () => (
  <div className={styles.titlesContainer}>
    <h1>Weather search</h1>
    <p>Find out weather, temperature and other conditions</p>
  </div>
);

export default Titles;
