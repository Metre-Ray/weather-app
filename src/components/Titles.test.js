import React from "react";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, render, mount } from "enzyme";
import Titles from './Titles';

Enzyme.configure({ adapter: new Adapter() });

// npm test -- -u   - для обновления снэпшота

describe("Titles", () => {
  test("match shallow snapshot", () => {
    const wrapper = shallow(<Titles />);    // не заходит "внутрь" дочерних компонент, т.е., например, <Home/> останется <Home/> в снэпшоте
    expect(wrapper).toMatchSnapshot();
  });
  test("match render snapshot", () => {     // заходит "внутрь" дочерних компонент и отрисовывает их
    const wrapper = render(<Titles />);
    expect(wrapper).toMatchSnapshot();
  });
  test("match mount snapshot", () => {     // как shallow, но есть все хуки жизненного цикла
    const wrapper = mount(<Titles />);
    expect(wrapper).toMatchSnapshot();
  });
})
