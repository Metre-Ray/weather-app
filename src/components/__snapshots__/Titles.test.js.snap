// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titles match mount snapshot 1`] = `
<Titles>
  <div
    className="titlesContainer"
  >
    <h1>
      Weather search
    </h1>
    <p>
      Find out weather, temperature and other conditions
    </p>
  </div>
</Titles>
`;

exports[`Titles match render snapshot 1`] = `
<div
  class="titlesContainer"
>
  <h1>
    Weather search
  </h1>
  <p>
    Find out weather, temperature and other conditions
  </p>
</div>
`;

exports[`Titles match shallow snapshot 1`] = `
<div
  className="titlesContainer"
>
  <h1>
    Weather search
  </h1>
  <p>
    Find out weather, temperature and other conditions
  </p>
</div>
`;
